File: Solutions/27. Remove Element.py
# 27. Remove Element

# 1.
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        while val in nums:
            nums.remove(val)
        return len(nums)
# 2.
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        res = 0
        while val in nums:
            nums.pop(nums.index(val))
            nums.append('_')
            res += 1
        return len(nums) - res

File: LeetCode (en comments)/27. Remove Element.cpp
#include <vector>

class Solution {
public:
    int removeElement(std::vector<int>& nums, int val) {
        int res = 0;
        while (std::find(nums.begin(), nums.end(), val) != nums.end()) {
            nums.erase(std::find(nums.begin(), nums.end(), val));
            nums.push_back('_');
            res++;
        }
        return nums.size() - res;
    }
};
